plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImagePage.java

```diff
@@ -81,6 +84,68 @@ protected DeployImagePage(IWizard wizard, IDeployImagePageModel model) {
 		this.model = model;
 	}
 
+	/**
+	 * Callback that gets called when this page is going to be deactivated. This
+	 * is the chance to do a one-time remote call to the selected Docker daemon
+	 * to check if an image with the given name exists in the local cache or in
+	 * the remote registry (Docker Hub). This kind of validation is a
+	 * long-running process, so it should not be performed during the field
+	 * validation.
+	 * 
+	 * @param progress
+	 *            the direction that the wizard is moving: backwards/forwards
+	 * @param event
+	 *            the page changing event that may be use to veto the change
+	 * @param dbc
+	 *            the current data binding context
+	 */
+	protected void onPageWillGetDeactivated(Direction progress, PageChangingEvent event, DataBindingContext dbc) {
+		
+		/**
+		 * Inner class to perform the image search in the selected Docker daemon cache or on the remote registry. 
+		 */
+		class ImageValidatorJob extends Job {
+
+			public ImageValidatorJob(String name) {
+				super(name);
+			}
+
+			private boolean imageExistsLocally = false;
+			private boolean imageExistsRemotely = false;
+			
+			@Override
+			protected IStatus run(IProgressMonitor monitor) {
+				this.imageExistsLocally = model.imageExistsLocally(model.getImage());
+				if(!imageExistsLocally) {
+					this.imageExistsRemotely = model.imageExistsRemotely(model.getImage());
+				}
+				return Status.OK_STATUS;
+			}
+			
+			
+		}
+		final ImageValidatorJob imageValidator = new ImageValidatorJob("Looking-up the selected Docker image...");
+		try {
+			WizardUtils.runInWizard(imageValidator, getContainer());
+		} catch (InvocationTargetException | InterruptedException e) {
+			final String message = "Failed to verify if an image named '" + model.getImage() + "' exists.";
+			MessageDialog.openError(getShell(), "Error", message);
+			OpenShiftUIActivator.getDefault().getLogger().logError(message, e);
+		}
+		if (imageValidator.imageExistsLocally) {
+			return;
+		} else if (!imageValidator.imageExistsLocally && imageValidator.imageExistsRemotely) {
+			if (!MessageDialog.openConfirm(getShell(), "Missing Docker Image", "The Docker image named '"
+					+ model.getImage() + "' will need to be pulled before opening the next page.")) {
+				event.doit = false;
+			}
+		} else {
```

Comment:
We will get here if user cancels the job. WizardUtils.runInWizard in this case returns IStatus.CANCEL, event.doit should be set to false, but error message dialog should not be opened.